units/src/parse.rs

```diff
@@ -214,19 +214,20 @@ pub fn hex_u32_prefixed(s: &str) -> Result<u32, PrefixedHexError> {
 ///
 /// # Errors
 ///
-/// - If input string contains a `0x` (or `0X`) prefix.
-/// - If input string is not a valid hex encoding of a `u32`.
+/// - If the input string contains a `0x` (or `0X`) prefix.
+/// - If the input string is not a valid hex encoding of a `u32`.
 pub fn hex_u32_unprefixed(s: &str) -> Result<u32, UnprefixedHexError> {
     let checked = hex_check_unprefixed(s)?;
     Ok(hex_u32_unchecked(checked)?)
 }
 
-/// Parses a `u32` from a hex string.
+/// Parses a `u32` from an unprefixed hex string without first checking for a prefix.
 ///
 /// # Errors
 ///
-/// - If input string is not a valid hex encoding of a `u32`.
-/// - With `InvalidDigit` due to the `x` if there is a prefix.
+/// - If the input string contains a `0x` (or `0X`) prefix an `InvalidDigit` error is returned due
+///   to the `x`.
+/// - If the input string is not a valid hex encoding of a `u32`.
```

Comment:
The improved clarity is good but the introduced line wrapping is a bit ugly, can we make it more terse? Perhaps

```rust
/// - If the input string contains a `0x` (or `0X`) prefix, returns `InvalidDigit` due to the `x`.
```

Reply:
Thanks, I have used your suggestion